sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -395,9 +395,9 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    * @since 2.0.0
    */
   override def dropTempView(viewName: String): Boolean = {
-    sparkSession.sessionState.catalog.getTempView(viewName).exists { viewDef =>
+    sparkSession.sessionState.catalog.getTempView(viewName).exists { _ =>
       sparkSession.sharedState.cacheManager.uncacheQuery(
-        sparkSession, viewDef, cascade = false)
+        sparkSession.table(viewName), cascade = false)
```

Comment:
Shall we add an assert into `uncacheQuery` to make sure all passed in logical plans are analyzed? e.g.,

```scala
def uncacheQuery(
      spark: SparkSession,
      plan: LogicalPlan,
      cascade: Boolean,
      blocking: Boolean = false): Unit = {
    assert(plan.analyzed, "the plan input to `uncacheQuery` should be analyzed")
   ...
}
```

Reply:
I'm not totally sure about this. It seems common that a temporary view is dropped _after_ its dependent view/table is already gone, so this assertion will fail in those cases, so this assertion will fail.

Another approach is to ignore the possible non-fatal errors during query analysis and uncache, [similar to what's done in `DropTableCommand`](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala#L239).